### PR TITLE
Add a new keyboard: yoichiro/lunakey_mini

### PR DESCRIPTION
**Pull request by yoichiro**
This pull request adds a new keyboard designed by @yoichiro to the QMK Firmware.

## Description

I sent this pull request to add a new keyboard named "Lunakey Mini" to the QMK Firmware. Because this keyboard was designed by @yoichiro and I intend to add some other keyboards in near future, I defined the folder structure like `keyboards/yoichiro/lunakey_mini`.

And, I created the simple default key map definition only. I will add other key maps, if they are needed after releasing it..

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by yoichiro** [comment]
Could someone review this pull request? Thank you in advance!

**Comment by Dbroqua** [comment]
Hello,

I suppose reviewers are in Week-end and in few hours/days you're PR will be reviewed.

You just need to wait ^^.

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Just a heads up, all of us collaborators do this stuff in our free time. So it make take a bit for us to get to a PR to review it.  Additionally, this entire year has been very exhausting for many of us, which is lead to a slowdown in reviews, too.  

So if getting stuff reviewed quickly is important, and you have spare time you're willing to contribute, you can review other PRs.  While approvals from collaborators may carry more weight, as long as actually effort was put into it, we value reviews from any contributor willing to put in the time. 

But this isn't required, by any means.  It's just something that would help everyone, and make the process go just a bit faster.

**Comment by yoichiro** [comment]
@drashna Thank you for the review and the important advice. I could understand the situation of this project. I intend to start reviewing for other PRs as possible.

File: keyboards/yoichiro/lunakey_mini/config.h [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0x5954  // "YT" - Yoichiro Tanaka
+#define PRODUCT_ID   0x0001
+#define DEVICE_VER   0x0001
+#define MANUFACTURER yoichiro
+#define PRODUCT      yoichiro Lunakey Mini
```
Comment (fauxpark) [inline keyboards/yoichiro/lunakey_mini/config.h:27]:
```suggestion
#define PRODUCT      Lunakey Mini
```

File: keyboards/yoichiro/lunakey_mini/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+  [_ADJUST] = LAYOUT(
+//+--------+--------+--------+--------+--------+--------+                 +--------+--------+--------+--------+--------+--------+
+   RESET,   _______, _______, _______, _______, _______,                   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   \
+//+--------+--------+--------+--------+--------+--------+                 +--------+--------+--------+--------+--------+--------+
+   RGB_TOG, RGB_HUI, RGB_SAI, RGB_VAI, _______, _______,                   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  \
+//+--------+--------+--------+--------+--------+--------+                 +--------+--------+--------+--------+--------+--------+
+   RGB_MOD, RGB_HUD, RGB_SAD, RGB_VAD, _______, _______,                   _______, _______, _______, _______, _______, _______, \
+//+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+
+                              KC_LGUI, KC_LALT, LOWER,   KC_SPC,  KC_ENT,  RAISE,   KC_RALT, KC_RGUI                             \
```
Comment (fauxpark) [inline keyboards/yoichiro/lunakey_mini/keymaps/default/keymap.c:78]:
```suggestion
  [_QWERTY] = LAYOUT(
//+--------+--------+--------+--------+--------+--------+                 +--------+--------+--------+--------+--------+--------+
   KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,                      KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_BSPC,
//+--------+--------+--------+--------+--------+--------+                 +--------+--------+--------+--------+--------+--------+
   KC_LCTL, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,                      KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
//+--------+--------+--------+--------+--------+--------+                 +--------+--------+--------+--------+--------+--------+
   KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,                      KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,
//+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+
                              KC_LGUI, KC_LALT, LOWER,   KC_SPC,  KC_ENT,  RAISE,   KC_RALT, KC_RGUI
//+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+
  ),

  [_LOWER] = LAYOUT(
//+--------+--------+--------+--------+--------+--------+                 +--------+--------+--------+--------+--------+--------+
   KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,                      KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC,
//+--------+--------+--------+--------+--------+--------+                 +--------+--------+--------+--------+--------+--------+
   KC_LCTL, _______, _______, _______, _______, _______,                   KC_LEFT, KC_DOWN, KC_UP,   KC_RIGHT,_______, _______,
//+--------+--------+--------+--------+--------+--------+                 +--------+--------+--------+--------+--------+--------+
   KC_LSFT, _______, _______, _______, _______, _______,                   _______, _______, _______, _______, _______, _______,
//+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+
                              KC_LGUI, KC_LALT, LOWER,   KC_SPC,  KC_ENT,  RAISE,   KC_RALT, KC_RGUI
//+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+
  ),

  [_RAISE] = LAYOUT(
//+--------+--------+--------+--------+--------+--------+                 +--------+--------+--------+--------+--------+--------+
   KC_ESC,  KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC,                   KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_BSPC,
//+--------+--------+--------+--------+--------+--------+                 +--------+--------+--------+--------+--------+--------+
   KC_LCTL, _______, _______, _______, _______, _______,                   KC_MINS, KC_EQL,  KC_LCBR, KC_RCBR, KC_PIPE, KC_GRV,
//+--------+--------+--------+--------+--------+--------+                 +--------+--------+--------+--------+--------+--------+
   KC_LSFT, KC_TILD, _______, _______, _______, _______,                   KC_UNDS, KC_PLUS, KC_LBRC, KC_RBRC, KC_BSLS, KC_TILD,
//+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+
                              KC_LGUI, KC_LALT, LOWER,   KC_SPC,  KC_ENT,  RAISE,   KC_RALT, KC_RGUI
//+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+
  ),

  [_ADJUST] = LAYOUT(
//+--------+--------+--------+--------+--------+--------+                 +--------+--------+--------+--------+--------+--------+
   RESET,   _______, _______, _______, _______, _______,                   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,
//+--------+--------+--------+--------+--------+--------+                 +--------+--------+--------+--------+--------+--------+
   RGB_TOG, RGB_HUI, RGB_SAI, RGB_VAI, _______, _______,                   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
//+--------+--------+--------+--------+--------+--------+                 +--------+--------+--------+--------+--------+--------+
   RGB_MOD, RGB_HUD, RGB_SAD, RGB_VAD, _______, _______,                   _______, _______, _______, _______, _______, _______,
//+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+
                              KC_LGUI, KC_LALT, LOWER,   KC_SPC,  KC_ENT,  RAISE,   KC_RALT, KC_RGUI
```

File: keyboards/yoichiro/lunakey_mini/readme.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Lunakey Mini
+
+![lunakey_mini](https://i.imgur.com/b1oUINi.jpg)
+
+*This is a 40% split keyboard which has 3x6 vertically staggered keys and 4 thumbnail keys.*
```
Comment (fauxpark) [inline keyboards/yoichiro/lunakey_mini/readme.md:5]:
```suggestion
This is a 40% split keyboard which has 3x6 vertically staggered keys and 4 thumbnail keys.
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/yoichiro/lunakey_mini/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  _ADJUST,
+};
+
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
+  LOWER,
+  RAISE,
+  ADJUST,
+};
+
```
Comment (zvecr) [inline keyboards/yoichiro/lunakey_mini/keymaps/default/keymap.c:31]:
```suggestion
```

File: keyboards/yoichiro/lunakey_mini/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+            if (record->event.pressed) {
+                layer_on(_ADJUST);
+            } else {
+                layer_off(_ADJUST);
+            }
+            return false;
+            break;
+    }
+    return true;
+}
```
Comment (zvecr) [inline keyboards/yoichiro/lunakey_mini/keymaps/default/keymap.c:121]:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
  return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
}
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/yoichiro/lunakey_mini/config.h (hunk 1 above)
Comment (yoichiro) [reply keyboards/yoichiro/lunakey_mini/config.h:27]: @fauxpark Thank you for the review comment. However, when I read the documentation of VIA, I found the following sentence:

> It is highly recommended that the value of PRODUCT is changed to be a combination of designer/vendor and the device name.

https://caniusevia.com/docs/configuring_qmk/#change-product

Therefore, I wrote the PRODUCT value as `MyName ProductName`. Should I remove my name and remain only the product name?

File: keyboards/yoichiro/lunakey_mini/readme.md (hunk 3 above)
Comment (yoichiro) [reply keyboards/yoichiro/lunakey_mini/readme.md:5]: @fauxpark Fix at 8d9f5f330. Thanks!

File: keyboards/yoichiro/lunakey_mini/keymaps/default/keymap.c (hunk 5 above)
Comment (yoichiro) [reply keyboards/yoichiro/lunakey_mini/keymaps/default/keymap.c]: @zvecr Thank you for the suggestion! I fix the code according to your suggestion at 0058d7694.

File: keyboards/yoichiro/lunakey_mini/config.h (hunk 1 above)
Comment (yoichiro) [reply keyboards/yoichiro/lunakey_mini/config.h]: @fauxpark I tried to find the product name like `DesignerName ProductName`, but I could not find such cases. I accept your suggestion and fix at 08df382cb.

**Comment by yoichiro** [comment]
@fauxpark @zvecr Thank you for the reviews! I committed them according to your suggestions. Could you review them again?

File: keyboards/yoichiro/lunakey_mini/readme.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
+# Lunakey Mini
+
+![lunakey_mini](https://i.imgur.com/b1oUINi.jpg)
```
Comment (noroadsleft) [inline keyboards/yoichiro/lunakey_mini/readme.md:3]: Swaps this for a smaller image.

```suggestion
![lunakey_mini](https://i.imgur.com/b1oUINil.jpg)
```

File: keyboards/yoichiro/lunakey_mini/lunakey_mini.h [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+/* This is a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define LAYOUT( \
```
Comment (noroadsleft) [inline keyboards/yoichiro/lunakey_mini/lunakey_mini.h:29]:
```suggestion
#define LAYOUT_split_3x6_4( \
```

File: keyboards/yoichiro/lunakey_mini/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+  _RAISE,
+  _ADJUST,
+};
+
+#define LOWER MO(_LOWER)
+#define RAISE MO(_RAISE)
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+  [_QWERTY] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/yoichiro/lunakey_mini/keymaps/default/keymap.c:30]:
```suggestion
  [_QWERTY] = LAYOUT_split_3x6_4(
```

File: keyboards/yoichiro/lunakey_mini/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+//+--------+--------+--------+--------+--------+--------+                 +--------+--------+--------+--------+--------+--------+
+   KC_LCTL, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,                      KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
+//+--------+--------+--------+--------+--------+--------+                 +--------+--------+--------+--------+--------+--------+
+   KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,                      KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,
+//+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+
+                              KC_LGUI, KC_LALT, LOWER,   KC_SPC,  KC_ENT,  RAISE,   KC_RALT, KC_RGUI
+//+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+
+  ),
+
+  [_LOWER] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/yoichiro/lunakey_mini/keymaps/default/keymap.c:42]:
```suggestion
  [_LOWER] = LAYOUT_split_3x6_4(
```

File: keyboards/yoichiro/lunakey_mini/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+//+--------+--------+--------+--------+--------+--------+                 +--------+--------+--------+--------+--------+--------+
+   KC_LCTL, _______, _______, _______, _______, _______,                   KC_LEFT, KC_DOWN, KC_UP,   KC_RIGHT,_______, _______,
+//+--------+--------+--------+--------+--------+--------+                 +--------+--------+--------+--------+--------+--------+
+   KC_LSFT, _______, _______, _______, _______, _______,                   _______, _______, _______, _______, _______, _______,
+//+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+
+                              KC_LGUI, KC_LALT, LOWER,   KC_SPC,  KC_ENT,  RAISE,   KC_RALT, KC_RGUI
+//+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+
+  ),
+
+  [_RAISE] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/yoichiro/lunakey_mini/keymaps/default/keymap.c:54]:
```suggestion
  [_RAISE] = LAYOUT_split_3x6_4(
```

File: keyboards/yoichiro/lunakey_mini/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+//+--------+--------+--------+--------+--------+--------+                 +--------+--------+--------+--------+--------+--------+
+   KC_LCTL, _______, _______, _______, _______, _______,                   KC_MINS, KC_EQL,  KC_LCBR, KC_RCBR, KC_PIPE, KC_GRV,
+//+--------+--------+--------+--------+--------+--------+                 +--------+--------+--------+--------+--------+--------+
+   KC_LSFT, KC_TILD, _______, _______, _______, _______,                   KC_UNDS, KC_PLUS, KC_LBRC, KC_RBRC, KC_BSLS, KC_TILD,
+//+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+
+                              KC_LGUI, KC_LALT, LOWER,   KC_SPC,  KC_ENT,  RAISE,   KC_RALT, KC_RGUI
+//+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+
+  ),
+
+  [_ADJUST] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/yoichiro/lunakey_mini/keymaps/default/keymap.c:66]:
```suggestion
  [_ADJUST] = LAYOUT_split_3x6_4(
```

File: keyboards/yoichiro/lunakey_mini/info.json [hunk 12]
```diff
@@ -0,0 +1,56 @@
+{
+    "keyboard_name": "Lunakey Mini",
+    "url": "",
+    "maintainer": "qmk",
+    "width": 14.75,
+    "height": 5,
+    "layouts": {
+        "LAYOUT": {
```
Comment (noroadsleft) [inline keyboards/yoichiro/lunakey_mini/info.json:8]:
```suggestion
        "LAYOUT_split_3x6_4": {
```

File: keyboards/yoichiro/lunakey_mini/info.json [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+                {"label":"R20", "x":8.75, "y":2.5},
+                {"label":"R21", "x":9.75, "y":2.25},
+                {"label":"R22", "x":10.75, "y":2},
+                {"label":"R23", "x":11.75, "y":2.5},
+                {"label":"R24", "x":12.75, "y":3.25},
+                {"label":"R25", "x":13.75, "y":3.5},
+                {"label":"R30", "x":7.75, "y":4},
+                {"label":"R31", "x":8.75, "y":3.75},
+                {"label":"R32", "x":9.75, "y":3.75},
+                {"label":"R33", "x":10.75, "y":4}]
```
Comment (noroadsleft) [inline keyboards/yoichiro/lunakey_mini/info.json:53]: This block needs to follow the order given by `lunakey_mini.h` – the same order that your `keymap.c` files are in. So `L00 ... L05`, then `R00 ... R05`, then `L10 ... L15`, and so on.

```suggestion
                {"label":"L00", "x":0, "y":1.5},
                {"label":"L01", "x":1, "y":1.25},
                {"label":"L02", "x":2, "y":0.5},
                {"label":"L03", "x":3, "y":0},
                {"label":"L04", "x":4, "y":0.25},
                {"label":"L05", "x":5, "y":0.5},
                {"label":"R00", "x":8.75, "y":0.5},
                {"label":"R01", "x":9.75, "y":0.25},
                {"label":"R02", "x":10.75, "y":0},
                {"label":"R03", "x":11.75, "y":0.5},
                {"label":"R04", "x":12.75, "y":1.25},
                {"label":"R05", "x":13.75, "y":1.5},
                {"label":"L10", "x":0, "y":2.5},
                {"label":"L11", "x":1, "y":2.25},
                {"label":"L12", "x":2, "y":1.5},
                {"label":"L13", "x":3, "y":1},
                {"label":"L14", "x":4, "y":1.25},
                {"label":"L15", "x":5, "y":1.5},
                {"label":"R10", "x":8.75, "y":1.5},
                {"label":"R11", "x":9.75, "y":1.25},
                {"label":"R12", "x":10.75, "y":1},
                {"label":"R13", "x":11.75, "y":1.5},
                {"label":"R14", "x":12.75, "y":2.25},
                {"label":"R15", "x":13.75, "y":2.5},
                {"label":"L20", "x":0, "y":3.5},
                {"label":"L21", "x":1, "y":3.25},
                {"label":"L22", "x":2, "y":2.5},
                {"label":"L23", "x":3, "y":2},
                {"label":"L24", "x":4, "y":2.25},
                {"label":"L25", "x":5, "y":2.5},
                {"label":"R20", "x":8.75, "y":2.5},
                {"label":"R21", "x":9.75, "y":2.25},
                {"label":"R22", "x":10.75, "y":2},
                {"label":"R23", "x":11.75, "y":2.5},
                {"label":"R24", "x":12.75, "y":3.25},
                {"label":"R25", "x":13.75, "y":3.5},
                {"label":"L30", "x":3, "y":4},
                {"label":"L31", "x":4, "y":3.75},
                {"label":"L32", "x":5, "y":3.75},
                {"label":"L33", "x":6, "y":4},
                {"label":"R30", "x":7.75, "y":4},
                {"label":"R31", "x":8.75, "y":3.75},
                {"label":"R32", "x":9.75, "y":3.75},
                {"label":"R33", "x":10.75, "y":4}
            ]
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
All these suggestions except for the last are oriented towards compatibility with the [Community Layouts](https://docs.qmk.fm/#/feature_layouts) feature.

File: keyboards/yoichiro/lunakey_mini/info.json (hunk 13 above)
Comment (yoichiro) [reply keyboards/yoichiro/lunakey_mini/info.json:53]: @noroadsleft Oh, Ok, I didn't know the policy. Thank you, I fix this.

**Comment by yoichiro** [comment]
@noroadsleft Thank you for the review and many suggestions! I fix them at eba3a84. Could you confirm them?

**Review by fauxpark — APPROVED** [review]

**Comment by yoichiro** [comment]
Why is the Travis CI still pending for a long time?

**Review by zvecr — APPROVED** [review]

**Comment by zvecr** [comment]
Thanks!

**Pull request merged**